internal/rest-client/flasharray_client.go

```diff
@@ -96,8 +99,10 @@ func NewRestClient(endpoint string, apitoken string, apiversion string, uagent s
 		return fa
 	}
 	fa.XAuthToken = res.Header().Get("x-auth-token")
+	fa.XRequestID = res.Header().Get("X-Request-ID")
 	fa.RestClient.SetHeader("x-auth-token", fa.XAuthToken)
 	fa.RestClient.SetHeader("User-Agent", FARestUserAgent+" ("+uagent+")")
+	fa.RestClient.SetHeader("X-Request-ID", fa.XRequestID)
```

Comment:
as we've set the request id to fa.RestClient in line 58. Do we still need to set it again here?

Reply:
yes, this is for the case where the FlashArray generates the request id, because one was not provided. It's also why we grab it from the response header too.